### PR TITLE
[CP-2109] [Center] Discrepancies in registering Matomo event different that download

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-2109]

**Description**

- [ ] getState function in async thunk actions is correctly typed
- [ ] redux selectors are used in components / prop drilling is reduce

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-2109]: https://appnroll.atlassian.net/browse/CP-2109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts [hunk 1]
```diff
@@ -44,6 +44,13 @@ export class AnalyticDataTrackerService implements AnalyticDataTrackerClass {
     event: TrackEvent,
     externalUsageDeviceOnly = true
   ): Promise<AxiosResponse | undefined> {
+    console.log(
+      this.trackingEnabled,
+      externalUsageDeviceOnly,
+      this.externalUsageDevice,
+      event.e_c
+    )
+
```
Comment (patryk-sierzega) [inline packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts:53]: this seems legit, but still is console.log :) Should it stay here?

**Review by patryk-sierzega — DISMISSED** [review]

File: packages/app/src/update/actions/start-update-os/start-update-os.action.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ export const startUpdateOs = createAsyncThunk<
     let state = getState()
     const batteryLevel = state.device.data?.batteryLevel ?? 0
     const deviceType = state.device.deviceType
+    const externalUsageDevice = state.device.externalUsageDevice
```
Comment (dkarski) [inline packages/app/src/update/actions/start-update-os/start-update-os.action.ts:58]: 🟢 
```suggestion
    const externalUsageDevice = state.device.externalUsageDevice ?? false
```

**Review by dkarski — APPROVED** [review]

File: packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts (hunk 1 above)
Comment (lkowalczyk87) [reply packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.ts]: resolved in 94eb632df33bbd05fa6bba41bcb9482d6d1cfcd5

File: packages/app/src/update/actions/start-update-os/start-update-os.action.ts (hunk 2 above)
Comment (lkowalczyk87) [reply packages/app/src/update/actions/start-update-os/start-update-os.action.ts]: resolved in 94eb632df33bbd05fa6bba41bcb9482d6d1cfcd5

**Review by patryk-sierzega — APPROVED** [review]

**Review by dkarski — APPROVED** [review]

**Pull request merged**